drivers/usbhid-ups.c

```diff
@@ -1422,7 +1422,8 @@ static void ups_status_set(void)
 		dstate_delinfo("input.transfer.reason");
 	}
 
-	if (ups_status & STATUS(ONLINE)) {
+	if ((ups_status & STATUS(ONLINE)) &&
+                !(ups_status & STATUS(DISCHRG))) {
```

Comment:
As mentioned by PR author and in later comments, this code change *probably* should not impact everyone silently, so needs a config toggle as a quirk for certain devices, to be set at users' discretion.